test/chain.js

```diff
@@ -62,24 +60,21 @@ metatests.test('for chain error', test => {
 metatests.test('for chain after fetch', test => {
   metasync
     .for([1, 2, 3, 4])
-    .map((item, cb) => cb(null, item * item))
-    .filter((item, cb) => cb(null, item > 5))
+    .map((item, cb) => process.nextTick(() => cb(null, item * item)))
+    .filter((item, cb) => process.nextTick(() => cb(null, item > 5)))
     .fetch((error, result, resume) => {
       test.error(error);
       test.strictSame(result, [9, 16]);
       resume(null, result);
-    })
-    .filter((item, cb) => {
-      cb(null, item > 10);
-    })
-    .map((item, cb) => {
-      cb(null, --item);
-    })
-    .fetch((error, result) => {
-      test.error(error);
-      test.strictSame(result, [15]);
       test.end();
     });
+  // .filter((item, cb) => process.nextTick(() => cb(null, item < 4)))
```

Comment:
Same here.

Reply:
This problem with `fetch after fetch` wasn't solved by me. Ask @nechaido.